napari/layers/_shapes_layer/shape_list.py

```diff
@@ -565,3 +540,120 @@ def inside(self, coord):
             return ordered_shapes[0]
         else:
             return None
+
+    def to_list(self, shape_type=None):
+        """Returns the vertex data assoicated with the shapes as a list
+        where each element of the list corresponds to one shape. Passing a
+        `shape_type` argument leads to only that particular `shape_type`
+        being returned.
+
+        Parameters
+        ----------
+        shape_type : str | None
+            String of shape shape_type, must be one of "{'line', 'rectangle',
+            'ellipse', 'path', 'polygon'}".
+
+        Returns
+        ----------
+        data : list
+            List of shape data where each element of the list is an
+            `np.ndarray` corresponding to one shape
+        """
+        if shape_type is None:
+            data = [s.data for s in self.shapes]
+        elif shape_type not in self._types.keys():
+            raise ValueError("""shape_type not recognized, must be one of
+                         "{'line', 'rectangle', 'ellipse', 'path',
+                         'polygon'}"
+                         """)
+        else:
+            cls = self._types[shape_type]
+            data = [s.data for s in self.shapes if type(s) == cls]
+        return data
+
+    def to_masks(self, mask_shape=None, shape_type=None):
+        """Returns N binary masks, one for each shape, embedded in an array of
+        shape mask_shape Passing a `shape_type` argument leads to only mask
+        from that particular `shape_type` being returned.
+
+        Parameters
+        ----------
+        mask_shape : np.ndarray | tuple | None
+            1x2 array of shape of mask to be generated. If non specified, takes
+            the max of all the vertiecs
+        shape_type : str | None
+            String of shape shape_type, must be one of "{'line', 'rectangle',
+            'ellipse', 'path', 'polygon'}".
+
+        Returns
+        ----------
+        masks : np.ndarray
+            NxMxP array where the is one binary mask of shape MxP for each of
```

Comment:
```suggestion
            NxMxP array where there is one binary mask of shape MxP for each of
```

You can also specify an array's shape in the "type": `masks : (N, M, P) array`

Reply:
fixed